Chapter1/2-HelloTriangle/Window.cs

```diff
@@ -160,33 +162,46 @@ protected override void OnRenderFrame(FrameEventArgs e)
             SwapBuffers();
 
             // And that's all you have to do for rendering! You should now see a yellow triangle on a black screen.
-            base.OnRenderFrame(e);
         }
 
         protected override void OnUpdateFrame(FrameEventArgs e)
         {
+            base.OnUpdateFrame(e);
+
             var input = KeyboardState;
 
             if (input.IsKeyDown(Keys.Escape))
             {
                 Close();
             }
-
-            base.OnUpdateFrame(e);
         }
 
         protected override void OnResize(ResizeEventArgs e)
         {
+            base.OnResize(e);
+
             // When the window gets resized, we have to call GL.Viewport to resize OpenGL's viewport to match the new size.
             // If we don't, the NDC will no longer be correct.
             GL.Viewport(0, 0, Size.X, Size.Y);
-            base.OnResize(e);
         }
 
-        // Now, for cleanup. This isn't technically necessary since C# and OpenGL will clean up all resources automatically when
-        // the program closes, but it's very important to know how anyway.
+        // Now, for cleanup.
+        // You should generally not do cleanup of opengl resources when exiting an application
+        // as that is handled by the driver and operating system when the application exits.
+        // 
+        // There are reasons to delete opengl resources but exiting the application is not one of them.
+        // This is provided here as a reference on how resoruce cleanup is done in opengl but
+        // should not be done when exiting the application.
+        //
+        // Places where cleanup is appropriate would be to delete textures that are no
+        // longer used for whatever reason (e.g. a new scene is loaded that doesn't use a texture).
+        // This would free up video ram (VRAM) that can be used for new textures.
+        //
+        // The comming chapters will not have this code.
         protected override void OnUnload()
         {
+            base.OnUnload();
```

Comment:
I think in `Unload` it makes sense to keep it at the end of the method, as you are unwinding all resources in the opposite order than `Load`

Reply:
At least in the case of Unload there's no actual code running in the supercall (outside of firing events) so I'd say the order doesn't really matter.

Reply:
I am aware, but for consistency reasons and to show newcomers